KestrelMock.Tests/MockTests.cs

```diff
@@ -218,6 +218,37 @@ public async Task KestralMock_works_with_Refit()
 
             Assert.Contains("world", helloWorld.Hello);
         }
+
+        [Theory]
+        [InlineData("starts/with/but_does_not_match_verb", "NotFound", "")]
+        [InlineData("starts/with/matches_put_method", "OK", "foo")]
+        [InlineData("/test/1234/xyz", "NotFound", "")]
+        [InlineData("/test/1234/xyz", "OK", "foo")]
+        [InlineData("/hello/world", "NotFound", "")]
+        [InlineData("/hello/world", "OK", "foo")]
+        [InlineData("api/estimate", "OK", "00000")]
+        [InlineData("api/estimate", "OK", "foo")]
+        public async Task KestralMock_matches_using_verb(string url, string statusCode, string body)
+        {
+            var client = _factory.CreateClient();
```

Comment:
just as an improvement suggestion: maybe to make this tests more pragmatic we could override the configuration of the mock, 
with an in-memory version of **MockSettings**, that we can change on tests, so that we don't have to rely on appsettings.json, and
the tests are more "readable" directly in the scope of the tests (without jumping to appsetting.json to verify how mock is configured)

```csharp
   var client = _factory
                .WithWebHostBuilder(builder =>
                    builder.ConfigureTestServices(services => 
                          //...
                          services.Configure<MockSettings>(customTestMockSettings) // instead of reading from file
```

or also maybe simplified, registering a in-memory version of appsettings in the contest of the test application factory

```csharp
 : WebApplicationFactory<WebService.Startup>
//...

public MockSettings TestMockSettings {get;set;} // though based on mutation... not the best
here in .ConfigureServices -->
services.Configure<MockSettings>(TestMockSettings);

//then in single tests just
_factory.TestMockSettings //modify as you like 
```

Reply:
yeah I had that same thought but figured first I would focus on getting the bug fixed. Definitely next step is to do your suggestion in another separate PR.